bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/IdentityTokenSupplier.java

```diff
@@ -23,73 +22,41 @@ public class IdentityTokenSupplier implements Serializable {
   private static final boolean license = true;
-  private long lastFetchTime;
-  private String cachedToken;
 
-  public IdentityTokenSupplier(String audience) {
-    this.audience = audience;
-    cachedToken = null;
-    lastFetchTime = 0;
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hashCode(cachedToken, lastFetchTime);
-  }
+  public static Optional<String> createIdentityToken(String audience) {
+    CloseableHttpClient httpClient = HttpClients.custom().setDefaultRequestConfig(config).build();
+    HttpGet httpGet =
+        new HttpGet(String.format(METADATA_VM_IDENTITY_URL, audience, format, license));
+    httpGet.addHeader(METADATA_FLAVOR_HEADER_KEY, METADATA_FLAVOR_HEADER_VALUE);
+    try {
+      CloseableHttpResponse response = httpClient.execute(httpGet);
+      if (response.getStatusLine().getStatusCode() == 200) {
 
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (!(o instanceof IdentityTokenSupplier)) {
-      return false;
-    }
-    IdentityTokenSupplier that = (IdentityTokenSupplier) o;
-    return Objects.equal(this.cachedToken, that.cachedToken)
-        && this.lastFetchTime == that.lastFetchTime;
-  }
-
-  public Optional<String> getIdentityToken() {
-    long currentTimeInMillis = System.currentTimeMillis();
-    if (cachedToken == null || currentTimeInMillis - lastFetchTime >= ttlInMillis) {
-      CloseableHttpClient httpClient = HttpClients.custom().setDefaultRequestConfig(config).build();
-      HttpGet httpGet =
-          new HttpGet(String.format(METADATA_VM_IDENTITY_URL, audience, format, license));
-      httpGet.addHeader(METADATA_FLAVOR_HEADER_KEY, METADATA_FLAVOR_HEADER_VALUE);
+        return Optional.of(
+            CharStreams.toString(new InputStreamReader(response.getEntity().getContent(), UTF_8)));
+      } else {
+        log.info(
+            "Unable to obtain identity token, response code: [{}]",
+            response.getStatusLine().getStatusCode());
+      }
+    } catch (IOException ex) {
+      log.info("Unable to obtain identity token", ex);
+    } finally {
       try {
-        CloseableHttpResponse response = httpClient.execute(httpGet);
-        if (response.getStatusLine().getStatusCode() == 200) {
-          cachedToken =
-              CharStreams.toString(new InputStreamReader(response.getEntity().getContent(), UTF_8));
-        } else {
-          cachedToken = null;
-          log.info(
-              "Unable to obtain identity token, response code: [{}]",
-              response.getStatusLine().getStatusCode());
-        }
-      } catch (IOException ex) {
-        log.info("Unable to obtain identity token", ex);
-        cachedToken = null;
-      } finally {
-        try {
-          Closeables.close(httpClient, true);
-        } catch (IOException e) {
-          // nothing to do
-        }
+        Closeables.close(httpClient, true);
+      } catch (IOException e) {
+        // nothing to do
```

Comment:
Should we add logs for this case well?

Reply:
Updated the fetch logic to use IdTokenProvider library